### PR TITLE
De-duplicate layer properties dialog code

### PR DESCRIPTION
**Pull request by nyalldawson**
Start a new class for common functionality across layer properties dialogs, and consolidate all duplicate metadata and style handling methods in there.

### CODE REVIEW

**Review by troopa81 — COMMENTED** [review]
Thank you for tackling this, it's a mess to maintain all this duplicated code

Did you consider using inheritance over aggreation for QgsLayerPropertiesGuiUtils ? I'm not really found of using qt signal/slot as callback, it makes debugging harder IMHO.

**Comment by nyalldawson** [comment]
@troopa81 
> Did you consider using inheritance over aggreation for QgsLayerPropertiesGuiUtils ? I'm not really found of using qt signal/slot as callback, it makes debugging harder IMHO.

Good move, done!

File: src/gui/raster/qgsrasterlayerproperties.cpp [hunk 1]
```diff
@@ -1609,83 +1600,17 @@ void QgsRasterLayerProperties::removeSelectedMetadataUrl()
...
-  {
-    //otherwise let the user know what went wrong
-    QMessageBox::information( this,
-                              tr( "Default Style" ),
-                              myMessage
-                            );
-  }
-}
 
 void QgsRasterLayerProperties::saveDefaultStyle()
```
Comment (troopa81) [inline src/gui/raster/qgsrasterlayerproperties.cpp:1604]: Shall we not mark the old methods as deprecated ?

File: src/app/pointcloud/qgspointcloudlayerproperties.h [hunk 2]
```diff
@@ -88,80 +87,35 @@ class QgsPointCloudClassificationStatisticsModel : public QAbstractTableModel
...
-     * \since QGIS 3.30
-     */
-    void loadStyle();
-
-    /**
-     * Saves a style when appriate button is pressed
-     *
-     * \since QGIS 3.30
-     */
-    void saveStyleAs();
```
Comment (troopa81) [inline src/app/pointcloud/qgspointcloudlayerproperties.h:126]: this method is part of the API, we need to keep it (and deprecate it)

File: src/app/pointcloud/qgspointcloudlayerproperties.h [hunk 3]
```diff
@@ -88,80 +87,35 @@ class QgsPointCloudClassificationStatisticsModel : public QAbstractTableModel
...
-     * \since QGIS 3.30
-     */
-    void loadDefaultStyle();
-
-    /**
-     * Saves the default style when appropriate button is pressed
-     *
-     * \since QGIS 3.30
-     */
-    void saveDefaultStyle();
```
Comment (troopa81) [inline src/app/pointcloud/qgspointcloudlayerproperties.h:112]: this method is part of the API, we need to keep it (and deprecate it)

**Review by troopa81 — COMMENTED** [review]
I think that old methods with different name from new one need to be deprecated but not removed

File: src/app/pointcloud/qgspointcloudlayerproperties.h (hunk 3 above)
Comment (nyalldawson) [reply src/app/pointcloud/qgspointcloudlayerproperties.h:112]: It's in app -- there is no stable API for this class

File: src/gui/raster/qgsrasterlayerproperties.cpp (hunk 1 above)
Comment (nyalldawson) [reply src/gui/raster/qgsrasterlayerproperties.cpp:1604]: Done!

**Pull request merged**